### PR TITLE
Pull Request for Issue1780: Implementing Zebra for BioXAS

### PR DESCRIPTION
**Pull request by helfrij**
This pull request handles adding the Zebra as the main control for the scaler and the Ge 32 element detector.  It involved making a new trigger source and some small tweaks to the 32 element detector class.  A new action was also created to handle ensuring that the 32 element detector is ready for acquisition since it can be slow some times.

### CODE REVIEW

File: source/beamline/AMDetectorTriggerSource.cpp [hunk 1]
```diff
@@ -40,13 +43,56 @@ void AMDetectorTriggerSource::setFailed(){
...
+	connect(detectorArmingMapper_, SIGNAL(mapped(QObject*)), this, SLOT(onDetectorArmed(QObject*)));
+
+	for(int x = 0, size = triggerSourceDetectors_.count(); x < size; x++)
+		triggerSourceDetectors_.at(x)->arm();
+}
+
+void AMArmedDetectorTriggerSource::addDetector(AMDetector *detector)
+{
+	triggerSourceDetectors_.append(detector);
+	connect(detector, SIGNAL(armed()), detectorArmingMapper_, SLOT(map(QObject*)));
```
Comment (helfrij) [inline source/beamline/AMDetectorTriggerSource.cpp]: Silly question @davidChevrier: is it possible for a detector to be armed before trigger(...) is called? In this case, the armed() signal could potentially be emitted before there is a mapping for it.

**Comment by dretrex** [comment]
This is ready for review now.  @iainworkman, could you take a look at it?

**Comment by helfrij** [comment]
@dretrex, it looks like this pull request is merging into masterBioXAS. Should I create a new pull request that merges into master?

**Comment by helfrij** [comment]
This pull request is being handled in #1854.

**Pull request closed without merging**